sw/device/tests/sim_dv/otp_ctrl_hw_cfg_test.c

```diff
@@ -0,0 +1,97 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <assert.h>
+#include <stdbool.h>
+
+#include "sw/device/lib/base/bitfield.h"
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_lc_ctrl.h"
+#include "sw/device/lib/dif/dif_otp_ctrl.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "lc_ctrl_regs.h"  // Generated.
+
+static dif_otp_ctrl_t otp;
+static dif_lc_ctrl_t lc;
+
+const test_config_t kTestConfig;
+const uint8_t NUM_DEVICE_ID = 8;
+
+/**
+ * Busy-wait until the DAI is done with whatever operation it is doing.
+ */
+static void wait_for_dai(void) {
+  while (true) {
+    dif_otp_ctrl_status_t status;
+    CHECK(dif_otp_ctrl_get_status(&otp, &status) == kDifOtpCtrlOk);
+    if (bitfield_bit32_read(status.codes, kDifOtpCtrlStatusCodeDaiIdle)) {
+      return;
+    }
+    LOG_INFO("Waiting for DAI...");
```

Comment:
Is this guaranteed to be a small time? Is it sensible to have some timeout?

Reply:
Thanks Matute, that is a good point. Usually it should only take a few clock cycles (maybe around 10) for read. I think we should be fine for that?